### DISCUSSION TITLE
Configurar SSL

### DISCUSSION THREAD

**Discussion by EFDevelopment** [HashLoad/horse#166] [Ideas]
Como configuro SSL como certificado auto assinado no Horse

**Comment by viniciussanchez** [comment]
O correto é que você não configure o certificado no Horse e sim no seu servidor de aplicação como IIS, Apache, Nginx.

**Reply by danielfrivas** [reply]
Vinicius, boa noite!
Tem algum vídeo mostrando como configurar o IIS e o certificado no IIS?
O que muda no Horse quando é utilizado o IIS?

**Reply by viniciussanchez** [reply]
Bom dia, no horse muda apenas a forma de criar o projeto, como mostra nos exemplos. E a diretiva de compilação. Quanto ao vídeo não tenho... mas no YT deve ter bastante

**Comment by EFDevelopment** [comment]
Consegui resolver em parte, conecto aplicação servidor Console com o certificado mais ao consumir o serviço apresenta o erro:

---------------------------
Debugger Exception Notification
---------------------------
Project Teste.exe raised exception class EIdSocketError with message 'Socket Error # 10054
Connection reset by peer.'.
---------------------------
Break   Continue   Help   
---------------------------

<codigo>
Const
  CertificatePassword = '1234';
  
  THorse.IOHandleSSL.CertFile:= '.\CertificadoI.crt'; //Certificado CRT
  THorse.IOHandleSSL.RootCertFile:= ''; //Host do Certificado
  THorse.IOHandleSSL.KeyFile:= '.\Certificado.key';  //chave key
  THorse.IOHandleSSL.OnGetPassword:= GetSSLPassWord;

Procedure TMainlyService.GetSSLPassWord (Var Password:String);
begin
  Password:= CertificatePassword;
end;

</codigo>

Utilizo a versao 1.0.2.21 do libeay e ssleay
e do open  1.1.1.4 ou 1.1..1.7 e mais atual do site todas retornaram o mesmo erro. Alguma ideia do que possa ser?

**Comment by EFDevelopment** [comment]
resolvido trocando https://localhost/ping para https://127.0.0.1/ping

Utilizando esta configuração:
```pascal
Const CertificatePassword = '1234';
THorse.IOHandleSSL.CertFile := '.\CertificadoI.crt'; //Certificado CRT
THorse.IOHandleSSL.RootCertFile := ''; //Host do Certificado
THorse.IOHandleSSL.KeyFile := '.\Certificado.key'; //chave key
THorse.IOHandleSSL.OnGetPassword := GetSSLPassWord;

Procedure TMainlyService.GetSSLPassWord (Var Password:String);
begin
    Password := CertificatePassword;
end;
```

**Comment by mariusmgarcia** [comment]
Bom dia, quando tento rodar minha aplicação com os parâmetros para usar o SLL recebo a seguinte msg:
Linha:
  THorse.IOHandleSSL.CertFile := '.\certif.crt';
  THorse.IOHandleSSL.KeyFile  := '.\certif.key';

Erro:
  [dcc32 Error] Unit1.pas(31): E2064 Left side cannot be assigned to

OBS.: No exemplo que vem junto está dando o mesmo erro na linha:
    KeyFile := leKey.Text;

Estou com a última versão do Horse instalado.

**Reply by viniciussanchez** [reply]
Bom dia. Você deve ter baixado os fontes em manutenção. Baixe sempre os fontes da última RELEASE liberada. Ou utilize o boss para efetuar a instalação

**Reply by mariusmgarcia** [reply]
Obrigado, era isso mesmo.

**Comment by kabiri** [comment]
Hello,
I'm using the latest release of version 3.1.6 and I'm encountering the following error:
[dcc32 Error] untMain.pas(81): E2064 Left side cannot be assigned to
on the line
THorse.IOHandleSSL.CertFile:= 'mySert.crt';

-----Edit------------
I’ve noticed that in the new version, it should be used like this:
THorse.IOHandleSSL.CertFile('mySert.crt');

libray : https://docwiki.embarcadero.com/RADStudio/Sydney/en/OpenSSL
